include/seqan3/align/detail/alignment.hpp

```diff
@@ -0,0 +1,130 @@
+
+namespace seqan3
+{
+
+template <typename t>
+concept bool aligned_sequence_concept = true;
+
+//! Alignment class.
+/*!
+    An alignment is a tuple of aligned sequences.
+*/
+template <typename first_t, typename ...remaining_ts>
+    requires aligned_sequence_concept<first_t> &&
+             (aligned_sequence_concept<remaining_ts> && ...)
+class alignment : public std::tuple<first_t, remaining_ts...>
+{
+public:
+    //! The alignment depth is the number of sequences contained.
+    std::size_t const depth = sizeof...(remaining_ts) + 1ul;
+
+    //! Constructor that allows to pass sequences directly.
+    constexpr alignment(first_t && v1, remaining_ts && ...v2)
+        : std::tuple<first_t, remaining_ts...>(v1, std::forward<remaining_ts>(v2)...){};
+};
+
+//! Alignment class template deduction as for tuple.
+template <typename ...ts>
+alignment(ts...) -> alignment<ts...>;
+
+namespace detail
+{
+// Helper for output stream operator.
+template <typename stream_t, typename tuple_t, size_t ...idx>
+void stream_alignment(stream_t & stream, tuple_t const & tuple, std::index_sequence<idx...> const & /**/)
+{
+    std::size_t const alignment_length = std::get<0>(tuple).size();
+    char buf[9];
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (std::size_t used_length = 0; used_length < alignment_length; used_length += 50)
+    {
+        // write header
+        std::snprintf(buf, 9, "%7lu ", used_length);
+        stream << std::endl << buf;
+        for (std::size_t col = 1; col <= 50 && col + used_length <= alignment_length; ++col)
+        {
+            if (col % 10 == 0)     stream << ':';
+            else if (col % 5 == 0) stream << '.';
+            else                   stream << ' ';
+        }
+
+        // write sequences
+        const char * indent = "        ";
+        stream << std::endl << indent << std::get<0>(tuple).substr(used_length, 50);
+        auto stream_f = [&]
+            (auto const & previous_sequence, auto const & aligned_sequence)
+        {
+            stream << std::endl << indent;
+            auto seq1 = previous_sequence.begin() + used_length;
+            auto seq2 = aligned_sequence.begin() + used_length;
+            for (auto it1 = seq1, it2 = seq2;
+                 it1 < previous_sequence.end() && it1 < seq1 + 50 &&
+                 it2 < aligned_sequence.end()  && it2 < seq2 + 50;
+                 ++it1, ++it2)
+            {
+                stream << (*it1 == *it2 ? '|' : ' ');
+            }
+            stream << std::endl << indent << aligned_sequence.substr(used_length, 50);
+        };
+        (stream_f(std::get<idx>(tuple), std::get<idx + 1>(tuple)), ...);
+        stream << std::endl;
+    }
+}
+
+} // namespace detail
+
+//! Formatted output stream of an alignment.
+template <typename stream_type, typename ...ts>
+stream_type & operator<<(stream_type & outstream, alignment<ts...> const & align)
+{
+    static_assert(sizeof...(ts) >= 2);
```

Comment:
Maybe add a message ? `static_assert(sizeof...(ts) >= 2, "oops, this needs to be more than two because...");`